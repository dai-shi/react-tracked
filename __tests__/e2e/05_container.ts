/* global page */

jest.setTimeout(15 * 1000);

describe('05_container', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(2) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(3) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(5) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(5) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(5) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(5) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(5) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(5) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(2)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    await page.click('body > #app > div:nth-child(6) > div > button:nth-child(3)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();
  });
});
