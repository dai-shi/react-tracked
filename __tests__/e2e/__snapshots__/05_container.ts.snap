// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`05_container should work with recorded events 1`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 10<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 12<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 2`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 14<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 16<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 3`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 18<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 20<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 4`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 22<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 24<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 5`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 26<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 28<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 6`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 30<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 32<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 7`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 34<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 36<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 8`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 38<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 9`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 42<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 44<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 10`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 6<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 8<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 11`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 50<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 52<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 12`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 54<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 56<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 13`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 58<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 60<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 14`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 62<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 64<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 15`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 66<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 68<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 16`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 70<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 72<div><span>Count: 2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 17`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 74<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 76<div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 18`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 78<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 80<div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 19`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 82<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 84<div><span>Count: -1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`05_container should work with recorded events 20`] = `
"
    <div id=\\"app\\"><h1>First container</h1><div>numRendered: 46<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 48<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><h1>Second container</h1><div>numRendered: 86<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div><div>numRendered: 88<div><span>Count: -2</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;
